### PR TITLE
pkg/*: issue service cert directly via cert manager api

### PR DESCRIPTION
**Pull request by shashankram**
**Description**:
This change allows SDS to directly use the IssueCertificate api
implemented by the certificate manager in use, instead of going
through MeshCatalog. This simplifies certificate issuance and code
duplication (such as catalog previously checking cert cache which
the IssueCertificate api implements within itself).

The tests are updated to reflect this change.

Will help #1965

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [X]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
`No`

### CODE REVIEW

File: DESIGN.md [hunk 1]
```diff
@@ -315,10 +315,6 @@ type MeshCataloger interface {
 	// ListEndpointsForService returns the list of provider endpoints corresponding to a service
 	ListEndpointsForService(service.MeshService) ([]endpoint.Endpoint, error)
 
-	// GetCertificateForService returns the SSL Certificate for the given service.
-	// This certificate will be used for service-to-service mTLS.
-	GetCertificateForService(service.MeshService) (certificate.Certificater, error)
```
Comment (draychev) [inline DESIGN.md:320]: Removing code feels so good! Thanks!

File: cmd/osm-controller/osm-controller.go [hunk 2]
```diff
@@ -209,7 +209,7 @@ func main() {
 	}
 
 	// Create and start the ADS gRPC service
-	xdsServer := ads.NewADSServer(meshCatalog, cfg.IsDebugServerEnabled(), osmNamespace, cfg)
+	xdsServer := ads.NewADSServer(meshCatalog, cfg.IsDebugServerEnabled(), osmNamespace, cfg, certManager)
```
Comment (draychev) [inline cmd/osm-controller/osm-controller.go:212]: Seems reasonable for the ADS server to have access to the certManager...

File: pkg/envoy/sds/response.go [hunk 3]
```diff
@@ -34,9 +34,9 @@ func NewResponse(catalog catalog.MeshCataloger, proxy *envoy.Proxy, request *xds
 		return nil, err
 	}
 
-	cert, err := catalog.GetCertificateForService(serviceForProxy)
+	cert, err := certManager.IssueCertificate(serviceForProxy.GetCommonName(), cfg.GetServiceCertValidityPeriod())
```
Comment (draychev) [inline pkg/envoy/sds/response.go:37]: 👍

**Review by draychev — APPROVED** [review]
It does indeed seem simpler to move `GetCertificateForService` out of the mesh catalog, and have SDS directly issue certs.
lgtm

**Review by snehachhabria — APPROVED** [review]

**Pull request merged**